Exception when using --single-instance

### Description

When a web site was nativefied with `--single-instance` option, attempt to launch 2nd instance result in an exception.

### Steps to reproduce issue

1. Create wrapper using `--single-instance` flag
2. In a terminal, launch 1st instance. All should be good so far
3. In another terminal, try launching 2nd instance
4. BANG! See exception below. The 1st instance was focused though.

### Details

>- Are you nativefying a *public* website?

Yes, `https://focus.nirvanahq.com/`

>- Feature request?

No

>- Full nativefier command used to build your app:

`nativefier --name "Nirvana" "https://focus.nirvanahq.com" --single-instance`

>- Version of Nativefier (run `nativefier --version`):

`7.6.7`

>- Version of node.js (run `node --version`):

`v8.10.0`

>- OS:

Linux Ubuntu 18.04, 4.15.0-29-generic, GNOME desktop

>- Error message / stack trace (if any):

```
$./nirvana-linux-x64/nirvana

module.js:545
    throw err;
    ^

Error: Cannot find module '../dialog'
    at Module._resolveFilename (module.js:543:15)
    at Function.Module._resolveFilename (/home/iblazhko/nirvana-linux-x64/resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.Module._load (module.js:473:25)
    at Module.require (module.js:586:17)
    at require (internal/module.js:11:18)
    at Object.get [as dialog] (/home/iblazhko/nirvana-linux-x64/resources/electron.asar/browser/api/exports/electron.js:11:16)
    at process.<anonymous> (/home/iblazhko/nirvana-linux-x64/resources/electron.asar/browser/init.js:54:31)
    at emitOne (events.js:116:13)
    at process.emit (events.js:211:7)
    at process.emit (/home/iblazhko/nirvana-linux-x64/resources/app/lib/webpack:/app/node_modules/source-map-support/source-map-support.js:454:1)
~ » ./nirvana-linux-x64/nirvana
module.js:545
    throw err;
    ^

Error: Cannot find module '../dialog'
    at Module._resolveFilename (module.js:543:15)
    at Function.Module._resolveFilename (/home/iblazhko/nirvana-linux-x64/resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.Module._load (module.js:473:25)
    at Module.require (module.js:586:17)
    at require (internal/module.js:11:18)
    at Object.get [as dialog] (/home/iblazhko/nirvana-linux-x64/resources/electron.asar/browser/api/exports/electron.js:11:16)
    at process.<anonymous> (/home/iblazhko/nirvana-linux-x64/resources/electron.asar/browser/init.js:54:31)
    at emitOne (events.js:116:13)
    at process.emit (events.js:211:7)
    at process.emit (/home/iblazhko/nirvana-linux-x64/resources/app/lib/webpack:/app/node_modules/source-map-support/source-map-support.js:454:1)
```

--- a/app/src/main.js
+++ b/app/src/main.js
@@ -127,21 +127,36 @@ if (appArgs.crashReporter) {
   });
 }
 
-app.on('ready', () => {
-  mainWindow = createMainWindow(appArgs, app.quit, setDockBadge);
-  createTrayIcon(appArgs, mainWindow);
+// quit if singleInstance mode and there's already another instance running
+let shouldQuit = appArgs.singleInstance && !app.requestSingleInstanceLock();
+if (shouldQuit) {
+  app.quit();
+} else {
+  app.on('second-instance', (event, commandLine, workingDirectory) => {
+    if (mainWindow) {
+      if (mainWindow.isMinimized()) {
+        mainWindow.restore()
+      }
+      mainWindow.focus()
+    }
+  });
 
-  // Register global shortcuts
-  if (appArgs.globalShortcuts) {
-    appArgs.globalShortcuts.forEach((shortcut) => {
-      globalShortcut.register(shortcut.key, () => {
-        shortcut.inputEvents.forEach((inputEvent) => {
-          mainWindow.webContents.sendInputEvent(inputEvent);
+  app.on('ready', () => {
+    mainWindow = createMainWindow(appArgs, app.quit, setDockBadge);
+    createTrayIcon(appArgs, mainWindow);
+
+    // Register global shortcuts
+    if (appArgs.globalShortcuts) {
+      appArgs.globalShortcuts.forEach((shortcut) => {
+        globalShortcut.register(shortcut.key, () => {
+          shortcut.inputEvents.forEach((inputEvent) => {
+            mainWindow.webContents.sendInputEvent(inputEvent);
+          });
         });
       });
-    });
-  }
-});
+    }
+  });
+}
 
 app.on('new-window-for-tab', () => {
   mainWindow.emit('new-tab');
@@ -160,24 +175,3 @@ app.on('login', (event, webContents, request, authInfo, callback) => {
     createLoginWindow(callback);
   }
 });
-
-if (appArgs.singleInstance) {
-  const shouldQuit = app.makeSingleInstance(() => {
-    // Someone tried to run a second instance, we should focus our window.
-    if (mainWindow) {
-      if (!mainWindow.isVisible()) {
-        // tray
-        mainWindow.show();
-      }
-      if (mainWindow.isMinimized()) {
-        // minimized
-        mainWindow.restore();
-      }
-      mainWindow.focus();
-    }
-  });
-
-  if (shouldQuit) {
-    app.quit();
-  }
-}

--- a/app/src/main.js
+++ b/app/src/main.js
@@ -128,16 +128,16 @@ if (appArgs.crashReporter) {
 }
 
 // quit if singleInstance mode and there's already another instance running
-let shouldQuit = appArgs.singleInstance && !app.requestSingleInstanceLock();
+const shouldQuit = appArgs.singleInstance && !app.requestSingleInstanceLock();
 if (shouldQuit) {
   app.quit();
 } else {
-  app.on('second-instance', (event, commandLine, workingDirectory) => {
+  app.on('second-instance', () => {
     if (mainWindow) {
       if (mainWindow.isMinimized()) {
-        mainWindow.restore()
+        mainWindow.restore();
       }
-      mainWindow.focus()
+      mainWindow.focus();
     }
   });
 

--- a/app/src/main.js
+++ b/app/src/main.js
@@ -134,7 +134,12 @@ if (shouldQuit) {
 } else {
   app.on('second-instance', () => {
     if (mainWindow) {
+      if (!mainWindow.isVisible()) {
+        // try
+        mainWindow.show();
+      }
       if (mainWindow.isMinimized()) {
+        // minimized
         mainWindow.restore();
       }
       mainWindow.focus();

